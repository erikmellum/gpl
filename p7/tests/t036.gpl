string str;
int i;
int size = 4;
rectangle rects[size];

on akey
{
    for (i = 0; i < size; i += 1)
    {
        rects[i].x = i * 100;
        rects[i].y = i * 100;
    }
}

on skey
{
    for (i = 0; i < size; i += 1)
    {
        str = "rects[" + i + "].x = " + rects[i].x;
        print(str);
        str = "rects[" + i + "].y = " + rects[i].y;
        print(str);
    }
}

