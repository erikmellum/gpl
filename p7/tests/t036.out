gpl.cpp::main()
  input file(tests/t036.gpl)
  random seed(42)
  read_keypresses_from_standard_input(true)
  dump_pixels(false)
  symbol_table(true)
  print_symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
int i 0
game_object rects[0]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
game_object rects[1]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
game_object rects[2]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
game_object rects[3]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
int size 4
string str ""
gpl.cpp::main() Calling window->initialize().
gpl.cpp::main() Passing control to window->main_loop().
gpl[20]: rects[0].x = 0
gpl[22]: rects[0].y = 0
gpl[20]: rects[1].x = 100
gpl[22]: rects[1].y = 100
gpl[20]: rects[2].x = 200
gpl[22]: rects[2].y = 200
gpl[20]: rects[3].x = 300
gpl[22]: rects[3].y = 300

Printing the symbol table from signal_handler()
int i 4
game_object rects[0]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
game_object rects[1]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(100),
    y = int(100)
  }
game_object rects[2]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(200),
    y = int(200)
  }
game_object rects[3]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(300),
    y = int(300)
  }
int size 4
string str "rects[3].y = 300"
